templates/templates.go

```diff
@@ -0,0 +1,11 @@
+package templates
+
+import "fmt"
+
+// TODO stop gap solution to get template content, once NSTemplateTier template is available, this should be removed.
+// Remove `templates` directory all togather.  Also remove `make generate` target
+// Remove gitignore entry - templates/bindata.go
+// Remove cleaning of templates/bindata.go from `make target`
+func GetTemplateContent(tmplName string) ([]byte, error) {
+	return Asset(fmt.Sprintf("templates/%s.yaml", tmplName))
```

Comment:
I tried template in one go_file as string.  But due to go_auto_format, the template_file content becoming invalid with one or other issue.  So switch to `make generate` way.  Anyway, This is temporary.

Reply:
The template stored as a string variable is used in the test of the template processor - there is no issue with formatting. Could you please shed some light on that formatting problem that you are facing?

Reply:
@MatousJobanek as you know this is temp.  Also, the effort is already made.  Now switching to something else will anyway need to revert the changes done.  So rather reverting now and replacing with something else.  Let's revert when we remove this temp solution.  I am less motivated to switch to string_version even if that's better option.  This all temp and should not be bother much as far as thing is in place :-)

Reply:
And I would vote for removing it now because:
* you know what should be removed because it's already part of the PR
* it will make the PR smaller
* the next PR that will introduce retrieval of `NsTemplateTier` will be easier, quicker and smaller

Reply:
That's why I capture all the changes in comment as what all to remove and it should be straight forward.  As I already mentioned, I am less motivated to make this changes as this is all temp stuff.

Reply:
Please, could you tell me why you don't want to remove it now? I don't see any benefit of keeping it here. Could you please tell me what the benefit is?

Reply:
@nurali-techie dropping the generation and using a string should be pretty easy to do unless we are missing something? We are probably spending more time discussing it than it would take to change the code in the PR :)

Reply:
> I tried template in one go_file as string. But due to go_auto_format, the template_file content becoming invalid with one or other issue. So switch to make generate way. Anyway, This is temporary.

@alexeykazakov I first started with string_way but as mentioned above, I got some issue in loading yaml_content.  So I thought to switch generation_way.  I thought that this is temp_solution so either solution would be fine.  Anyway, I will try to switch to string_way and try to fix issue which I got previously.

Reply:
> A template stored as a string variable is used in the test of the template processor - there is no issue with formatting. Could you please shed some light on that formatting problem that you are facing?

Below is the sort of error I am getting .. but anyway, let me try to find the fix.
```
$ go test ./pkg/controller/nstemplateset/ -v -count=1 -run TestCreateReconcile
=== RUN   TestCreateReconcile
=== RUN   TestCreateReconcile/ok
2019-09-18T09:45:24.926+0530	INFO	controller_nstemplateset	Reconciling NSTemplateSet	{"Request.Namespace": "toolchain-member", "Request.Name": "johnsmith"}
2019-09-18T09:45:24.927+0530	INFO	controller_nstemplateset	provisioning namespace	{"namespace": {"type":"dev","revision":"rev1"}}
2019-09-18T09:45:24.927+0530	ERROR	controller_nstemplateset	failed to provision user namespaces	{"Request.Namespace": "toolchain-member", "Request.Name": "johnsmith", "error": "failed to process template for namespace 'johnsmith-dev': unable to decode template: yaml: line 2: found a tab character that violates indentation", "errorVerbose": "yaml: line 2: found a tab character that violates indentation\nunable to decode template\ngithub.com/codeready-toolchain/member-operator/pkg/template.Processor.Process\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/template/processor.go:36\ngithub.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).ensureNamespaceResource\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:171\ngithub.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).ensureNamespace\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:155\ngithub.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).ensureUserNamespaces\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:140\ngithub.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).Reconcile\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:112\ngithub.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.TestCreateReconcile.func1\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller_test.go:104\ngithub.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.TestCreateReconcile.func2\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller_test.go:113\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:865\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1337\nfailed to process template for namespace 'johnsmith-dev'\ngithub.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).wrapErrorWithStatusUpdate\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:287\ngithub.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).ensureNamespaceResource\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:173\ngithub.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).ensureNamespace\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:155\ngithub.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).ensureUserNamespaces\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:140\ngithub.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).Reconcile\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:112\ngithub.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.TestCreateReconcile.func1\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller_test.go:104\ngithub.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.TestCreateReconcile.func2\n\t/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller_test.go:113\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:865\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1337"}
github.com/go-logr/zapr.(*zapLogger).Error
	/home/nvirani/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128
github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).Reconcile
	/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:114
github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.TestCreateReconcile.func1
	/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller_test.go:104
github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.TestCreateReconcile.func2
	/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller_test.go:113
testing.tRunner
	/usr/local/go/src/testing/testing.go:865
--- FAIL: TestCreateReconcile (0.00s)
    require.go:794: 
        	Error Trace:	nstemplateset_controller_test.go:105
        	            				nstemplateset_controller_test.go:113
        	Error:      	Received unexpected error:
        	            	yaml: line 2: found a tab character that violates indentation
        	            	unable to decode template
        	            	github.com/codeready-toolchain/member-operator/pkg/template.Processor.Process
        	            		/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/template/processor.go:36
        	            	github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).ensureNamespaceResource
        	            		/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:171
        	            	github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).ensureNamespace
        	            		/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:155
        	            	github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).ensureUserNamespaces
        	            		/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:140
        	            	github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).Reconcile
        	            		/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:112
        	            	github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.TestCreateReconcile.func1
        	            		/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller_test.go:104
        	            	github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.TestCreateReconcile.func2
        	            		/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller_test.go:113
        	            	testing.tRunner
        	            		/usr/local/go/src/testing/testing.go:865
        	            	runtime.goexit
        	            		/usr/local/go/src/runtime/asm_amd64.s:1337
        	            	failed to process template for namespace 'johnsmith-dev'
        	            	github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).wrapErrorWithStatusUpdate
        	            		/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:287
        	            	github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).ensureNamespaceResource
        	            		/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:173
        	            	github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).ensureNamespace
        	            		/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:155
        	            	github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).ensureUserNamespaces
        	            		/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:140
        	            	github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.(*ReconcileNSTemplateSet).Reconcile
        	            		/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller.go:112
        	            	github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.TestCreateReconcile.func1
        	            		/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller_test.go:104
        	            	github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset.TestCreateReconcile.func2
        	            		/home/nvirani/go/src/github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset/nstemplateset_controller_test.go:113
        	            	testing.tRunner
        	            		/usr/local/go/src/testing/testing.go:865
        	            	runtime.goexit
        	            		/usr/local/go/src/runtime/asm_amd64.s:1337
        	Test:       	TestCreateReconcile
    --- FAIL: TestCreateReconcile/ok (0.00s)
        testing.go:820: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
FAIL
FAIL	github.com/codeready-toolchain/member-operator/pkg/controller/nstemplateset	0.024s
```

Reply:
Done https://github.com/codeready-toolchain/member-operator/pull/53/commits/28b47a486c0941e633a725b07b0420f81debfa42 .. copy/paste yaml string from  `bindata.go` which was generated using `make generate`.